nifi-nar-bundles/nifi-extension-utils/nifi-hadoop-utils/src/main/java/org/apache/nifi/hadoop/SecurityUtil.java

```diff
@@ -51,7 +50,8 @@ public static synchronized UserGroupInformation loginKerberos(final Configuratio
         Validate.notNull(keyTab);
 
         UserGroupInformation.setConfiguration(config);
-        return UserGroupInformation.loginUserFromKeytabAndReturnUGI(principal.trim(), keyTab.trim());
+        UserGroupInformation.loginUserFromKeytab(principal.trim(), keyTab.trim());
```

Comment:
I think we should document in bold language precisely why this change is so critical.  That the method that was being used did not set the static loginContext in the UGI object meant that its utility once a relogin/expiry hits was nearly nil is critical and we should document this foreverever.

Reply:
And we should also probably in that comment explain why the ticket renewal threads to attempt to force explicit renewals could be problematic/increase chances of race conditions.  Specifically the subject within the UGI could be loggedout by our explicit renewal attempts while at the same time a hadoop operation occurring could kick off the Hadoop client to relogin but the subject would have been cleared/in an unexpected state.  The UGI class passes the subject to the underlying jdk kerb handling.

Reply:
I've added javadoc to several methods that are affected by this change.  Will update the PR with these changes.